builder/bundles.go

```diff
@@ -436,17 +436,51 @@ func genUpdateBundleSpecialFiles(chrootDir string, b *Builder) error {
 	return ioutil.WriteFile(filepath.Join(swupdDir, "format"), []byte(b.State.Mix.Format), 0644)
 }
 
-func installBundleToFull(packagerCmd []string, buildVersionDir string, bundle *bundle) error {
+func downloadRpms(packagerCmd, rpmList []string, baseDir string, maxRetries int) error {
+	var err error
+
+	if maxRetries <= 0 {
+		return errors.Errorf("maxRetries value <= 0 for RPM downloads")
+	}
+
+	args := merge(packagerCmd, "--installroot="+baseDir, "install", "--downloadonly")
+	args = append(args, rpmList...)
+
+	for retries := 0; retries < maxRetries; retries++ {
+		_, err = helpers.RunCommandOutputEnv(args[0], args[1:], []string{"LC_ALL=en_US.UTF-8"})
+		if err == nil {
+			return err
+		}
+
+		fmt.Printf("RPM download attempt %d failed. Retrying up to %d times.\n", retries+1, maxRetries)
+		if err := clearDNFCache(packagerCmd); err != nil {
```

Comment:
I added the clearDNFCache function to remove potentially corrupted data from the DNF cache.